pandas/io/pytables.py

```diff
@@ -4106,44 +4106,42 @@ def process_axes(self, obj, selection: Selection, columns=None) -> DataFrame:
         for axis, labels in self.non_index_axes:
             obj = _reindex_axis(obj, axis, labels, columns)
 
-        # apply the selection filters (but keep in the same order)
-        if selection.filter is not None:
-            for field, op, filt in selection.filter.format():
-
-                def process_filter(field, filt):
+        def process_filter(field, filt, op):
 
-                    for axis_name in obj._AXIS_ORDERS:
-                        axis_number = obj._get_axis_number(axis_name)
-                        axis_values = obj._get_axis(axis_name)
-                        assert axis_number is not None
+            for axis_name in obj._AXIS_ORDERS:
+                axis_number = obj._get_axis_number(axis_name)
+                axis_values = obj._get_axis(axis_name)
+                assert axis_number is not None
 
-                        # see if the field is the name of an axis
-                        if field == axis_name:
+                # see if the field is the name of an axis
+                if field == axis_name:
 
-                            # if we have a multi-index, then need to include
-                            # the levels
-                            if self.is_multi_index:
-                                filt = filt.union(Index(self.levels))
+                    # if we have a multi-index, then need to include
+                    # the levels
+                    if self.is_multi_index:
+                        filt = filt.union(Index(self.levels))
+                    takers = op(axis_values, filt)
+                    return obj.loc(axis=axis_number)[takers]
 
```

Comment:
nit - let's not remove this newline

other than that, looks good

Reply:
Whoops.      replaced newline, `git push`, let the tests begin.